### ISSUE TITLE
Update ExpressionSet to (Tree)SummarizedExperiment

### ISSUE THREAD

**Issue by jwokaty** [waldronlab/curatedMetagenomicData#172]
Just wanted to give an update on this, a `ExpressionSet` won't handle sparse matrices – so this would have to be coupled with an update to `SummarizedExperiment`. Will leave this alone for now.

_Originally posted by @schifferl in https://github.com/waldronlab/curatedMetagenomicData/issues/170#issuecomment-789993132_

@lwaldron mentioned TreeSummarizedExperiment might be a better fit; if it is, use that instead.

**Comment by lwaldron** [comment]
This is a convenience function for users that will load a matrix, filter metadata from combined_metadata, and return a SummarizedExperiment.

**Comment by LiNk-NY** [comment]
@schifferl Are we expecting the data (as downloaded from EH) to come in as an `ExpressionSet` (current state) or are we reworking the pipeline to provide `matrix`? 
Also should we be using the new constructor function `curatedMetagenomicData` for transitioning to `TreeSummarizedExperiment`? 
If so, Lucas, you would take care of it as part of the work on the constructor function. 

The helper would look just like the `TreeSummarizedExperiment` constructor function: 
```r
TreeSummarizedExperiment::TreeSummarizedExperiment(
    assays = list(Count = assay_data),
    rowData = row_data,
    colData = col_data,
    rowTree = row_tree,
    rowNodeLab = row_lab,
    colTree = col_tree)
```

**Comment by schifferl** [comment]
We should expect the data to arrive as matrix (a `base::matrix` or a `Matrix::dgTMatrix-class` matrix) from EH. No more `ExpressionSet` objects : )

And yes, the new constructor function `curatedMetagenomicData` should be used for transitioning to `SummarizedExperiment` or `TreeSummarizedExperiment`.

The draft code and example data from earlier today is attached ([curatedMetagenomicData.zip](https://github.com/waldronlab/curatedMetagenomicData/files/6133532/curatedMetagenomicData.zip)), but probably needs to be much closer to what you've done in `curatedTCGAData` and `SingleCellMultiModal`. The fewer dependencies the better.

**Comment by LiNk-NY** [comment]
@schifferl I think it would be good to create a branch and have the code live there until it is ready to merge to the main branch :smile:

**Comment by lwaldron** [comment]
See https://github.com/waldronlab/curatedMetagenomicData/tree/remodel branch.

**Comment by lwaldron** [comment]
Related to #165

**Comment by schifferl** [comment]
This issue is being closed with #188 and for the same reasons. The change to SummarizedExperiment has happened and the method will now evolve to return a TreeSummarizedExperiment – it is represented by this Trello card (https://trello.com/c/bU0kGfm2) rather than an issue.
